specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2023-08-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,2890 @@
+          },
+          "description": "The list of deleted configuration store."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of deleted configuration stores."
+        }
+      },
+      "description": "List of deleted configuration stores"
+    },
+    "ReplicaListResult": {
+      "description": "The result of a request to list replicas.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The collection value.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Replica"
+          }
+        },
+        "nextLink": {
+          "description": "The URI that can be used to request the next set of paged results.",
+          "type": "string"
+        }
+      }
+    },
+    "Replica": {
+      "description": "The replica resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The name of the replica.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The type of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "The location of the replica.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "Resource system metadata.",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/ReplicaProperties",
+          "description": "All replica properties.",
+          "x-ms-client-flatten": true,
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ReplicaProperties": {
+      "description": "All replica properties.",
+      "type": "object",
+      "properties": {
+        "endpoint": {
+          "description": "The URI of the replica where the replica API will be available.",
+          "type": "string",
+          "readOnly": true
+        },
+        "provisioningState": {
+          "description": "The provisioning state of the replica.",
+          "enum": [
+            "Creating",
+            "Succeeded",
+            "Deleting",
+            "Failed",
+            "Canceled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ReplicaProvisioningState",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "Snapshot": {
+      "description": "The snapshot resource.",
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

Should get the top-level properties (id, name, type) by $ref Resource definition in common types.

Reply:
Looks like using the Resource definition also brings in SystemData. This is a proxy for a data plane snapshot, which doesn't have the info needed to fill out all of the properties in the SystemData. Would SystemData need to be fully filled out if it is included?

For reference the data plane snapshot has these properties:

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/preview/2023-05-01-preview/appconfiguration.json#L1641C8-L1641C18

Reply:
For proxy resources, while RP is responsible filling in systemData properties, it should be fine. So, if some of these not present or filled in by RP, up to your service for proxy resources. But every resource in ARM has to follow ARM Proxy or Tracked model, instead of defining its own subset of properties.

Reply:
Tried to use the most recent common types here, but it triggered a breaking change:

https://github.com/Azure/azure-rest-api-specs/pull/25499/checks?check_run_id=17205675783

<html>
<body>
<!--StartFragment-->

Rule | Message
-- | --
❌ 1023 - TypeFormatChanged | The new version has a different format than the previous one.New: common-types/resource-management/v5/types.json#L14:9Old: common-types/resource-management/v2/types.json#L14:9

<!--EndFragment-->
</body>

Then found a lower version of common types that doesn't trigger the breaking change, but it adds a lintdiff warning:

https://github.com/Azure/azure-rest-api-specs/pull/25499/checks?check_run_id=17325508508

<html>
<body>
<!--StartFragment-->

Rule | Message
-- | -- 
⚠️ LatestVersionOfCommonTypesMustBeUsed | Use the latest version v5 of types.json.Location: Microsoft.AppConfiguration/preview/2023-08-01-preview/appconfiguration.json#L2670

<!--EndFragment-->
</body>
</html>

Is it ok to use the lower version of common types and ignore/suppress the lintdiff warning? Or is there a way to use to the most recent common types without triggering breaking changes?